### ISSUE TITLE
Calling super methods returning never does not contribute to CFA

### ISSUE THREAD

**Issue by Gerrit0** [microsoft/TypeScript#36147]
**TypeScript Version:**  3.7.2, Nightly

**Search Terms:**  super control flow analysis never, super returning never

**Expected behavior:** No error on `SuperThrowable.err`, or an error on both `SuperThrowable.err` and `SuperThrowable.ok`

**Actual behavior:** A function returning 'never' cannot have a reachable end point.(2534)

**Related Issues:** Quite a few regarding CFA, but none related to `super` vs `this`. Discovered from [this SO question](https://stackoverflow.com/q/59698803/7186598)

**Code**
```ts
class MyThrowable {
    throw(): never {
        throw new Error();
    }
}

class SuperThrowable extends MyThrowable {
    /// A function returning 'never' cannot have a reachable end point.(2534)
    err(msg: string): never {
        super.throw()
    }

    // Ok
    ok(): never {
        this.throw()
    }
}
```

<details><summary><b>Output</b></summary>

```ts
"use strict";
class MyThrowable {
    throw() {
        throw new Error();
    }
}
class SuperThrowable extends MyThrowable {
    /// A function returning 'never' cannot have a reachable end point.(2534)
    err(msg) {
        super.throw();
    }
    // Ok
    ok() {
        this.throw();
    }
}

```

</details>

<details><summary><b>Compiler Options</b></summary>

```json
{
  "compilerOptions": {
    "noImplicitAny": true,
    "strictNullChecks": true,
    "strictFunctionTypes": true,
    "strictPropertyInitialization": true,
    "strictBindCallApply": true,
    "noImplicitThis": true,
    "noImplicitReturns": true,
    "useDefineForClassFields": false,
    "alwaysStrict": true,
    "allowUnreachableCode": false,
    "allowUnusedLabels": false,
    "downlevelIteration": false,
    "noEmitHelpers": false,
    "noLib": false,
    "noStrictGenericChecks": false,
    "noUnusedLocals": false,
    "noUnusedParameters": false,
    "esModuleInterop": true,
    "preserveConstEnums": false,
    "removeComments": false,
    "skipLibCheck": false,
    "checkJs": false,
    "allowJs": false,
    "declaration": true,
    "experimentalDecorators": false,
    "emitDecoratorMetadata": false,
    "target": "ES2017",
    "module": "ESNext"
  }
}
```

</details>

**Playground Link:** [Provided](https://www.typescriptlang.org/play/index.html#code/MYGwhgzhAECyCeAVAFgJwPYHcwCMQFNoBvAWAChpLoAXNLACgEoAuaAO3wDd9VjyqBNOpnb4RAUVQZUTANz8qAX3LKy5UJBgBlAK4AHHigzY8hfAA9q+NgBMYCI1lwE+FKgHpP0AILQAZjpswNQAluhs0Kj41DqobCFsAObQAOQc3Kgp0MBgbGzo1NDIYNzQYJH4YMDFptDWNtB66AnUAHT0AEwArADMACyMCpQ8MgC2EImsENSoCYksohmuglQQ+jyttMZMQ9Cqu14A8gDWu+jHTKzpPMsrQiEQm8I7bpSqikA)

**Comment by nmain100** [comment]
I believe the following is related, as `never` returning functions impacting flow and `asserts` functions have the same limitations?

```ts
class C {
    a(b: any): asserts b is number {
        if (typeof b !== "number")
            throw new Error
    }
}

class D extends C {
    foo(b: any) {
        // Assertions require the call target to be an identifier or qualified name.(2776)
        super.a(b);
    }
}
```
[Playground](https://www.typescriptlang.org/play/?ssl=1&ssc=1&pln=13&pc=2#code/MYGwhgzhAEDC0G8BQ1XTACgEYC50DsBPASj0ggFMAnAFxi2gEsZ8BXAWy2sRTT8YBm0DDUIAHCgHshDAIQBeedABEbTtWXFefHdBoALKpIDu0fBVMBRKkarboAXyROkoctAAi0CgA8aFfAATGHhkPgFJSWwyImIeXVQAekToAEEoahpGSXwYKgoAR1ZGfL19CmhgMBAQPTAqAHMKGj1JaC4CJkCArIFGbkkqaCLqwX7AszB2CgA6DAAmAHZFgDYtBOgIVgkqGcwsYgBueycHIA)
